tests/test_settings.py

```diff
@@ -0,0 +1,99 @@
+        mock_open = mock.mock_open(read_data=b"[tool.screenpy]\nTIMEOUT = 500")
+
+        with mock.patch("pathlib.Path.open", mock_open):
+            settings = ScreenPySettings()
+
+        assert settings.TIMEOUT == 500
+
+    def test_env_overwrites_pyproject(self):
+        mock_open = mock.mock_open(read_data=b"[tool.screenpy]\nTIMEOUT = 500")
+        mock_env = {"SCREENPY_TIMEOUT": "1337"}
+
+        with mock.patch("pathlib.Path.open", mock_open):
+            with mock.patch.dict(os.environ, mock_env):
+                settings = ScreenPySettings()
+
+        assert settings.TIMEOUT == 1337
+
+    def test_init_overwrites_env(self):
+        mock_env = {"SCREENPY_TIMEOUT": "1337"}
+
+        with mock.patch.dict(os.environ, mock_env):
+            settings = ScreenPySettings(TIMEOUT=9001)
+
+        assert settings.TIMEOUT == 9001
+
+    def test_can_be_changed_at_runtime(self):
+        try:
+            screenpy_settings.TIMEOUT = 4
+        except TypeError:
+            msg = "ScreenPySettings could not be changed at runtime."
+            raise AssertionError(msg)
+
+
+class TestStdOutAdapterSettings:
+    def test_pyproject_overwrites_initial(self):
+        mock_open = mock.mock_open(
+            read_data=b"[tool.screenpy.stdoutadapter]\nINDENT_SIZE = 500"
+        )
+
+        with mock.patch("pathlib.Path.open", mock_open):
+            settings = StdOutAdapterSettings()
+
+        assert settings.INDENT_SIZE == 500
+
+    def test_env_overwrites_pyproject(self):
+        mock_open = mock.mock_open(
+            read_data=b"[tool.screenpy.stdoutadapter]\nINDENT_SIZE = 500"
+        )
+        mock_env = {"SCREENPY_STDOUTADAPTER_INDENT_SIZE": "1337"}
+
+        with mock.patch("pathlib.Path.open", mock_open):
+            with mock.patch.dict(os.environ, mock_env):
+                settings = StdOutAdapterSettings()
+
+        assert settings.INDENT_SIZE == 1337
+
+    def test_init_overwrites_env(self):
+        mock_env = {"SCREENPY_STDOUTADAPTER_INDENT_SIZE": "1337"}
+
+        with mock.patch.dict(os.environ, mock_env):
+            settings = StdOutAdapterSettings(INDENT_SIZE=9001)
+
+        assert settings.INDENT_SIZE == 9001
+
+    def test_can_be_changed_at_runtime(self):
+        try:
+            stdout_adapter_settings.INDENT_CHAR = "?"
+        except TypeError:
+            msg = "StdOutAdapterSettings could not be changed at runtime."
+            raise AssertionError(msg)
+
+
+class TestCombo:
+    def test_can_set_multiple_tools(self):
+        test_data = (
+            b"[tool.screenpy]\nTIMEOUT = 500"
+            b"\n\n[tool.screenpy.stdoutadapter]\nINDENT_SIZE = 500"
+        )
+        mock_open = mock.mock_open(read_data=test_data)
+
+        with mock.patch("pathlib.Path.open", mock_open):
+            base_settings = ScreenPySettings()
+            adapter_settings = StdOutAdapterSettings()
+
+        assert base_settings.TIMEOUT == 500
+        assert adapter_settings.INDENT_SIZE == 500
```

Comment:
Do we need a test for multiple when set in env?   or is that overkill?

Reply:
This was mostly a test for the "find pyproject.toml settings" function. Maybe i should actually have direct tests for that one...?

I think i'll rework this.